lib/internal/url.js

```diff
@@ -710,9 +635,8 @@ class URL {
     };
     const ctx = this[context];
     // https://url.spec.whatwg.org/#url-serializing
-    let ret = ctx.scheme;
-    if (ctx.host !== null) {
-      ret += '//';
+    let ret = ctx.protocol + '//';
+    if (ctx.host !== '') {
```

Comment:
1. This no longer matches the spec, and matters in the case of say `"javascript:void 0"`.
2. This whole function is super similar to `constructHref` earlier. Can they be combined?

Reply:
You're right. I'm updating the pull request with the correct implementation. Added couple of todos to be done after this pull request is merged (for simplifying the codebase).